datadriven/src/test_data_reader.rs

```diff
@@ -34,6 +38,7 @@ impl<'a> TestDataReader<'a> {
 
             let mut pos = line.unwrap().0;
             let mut line = line.unwrap().1.to_string();
+            self.emit(&line);
```

Comment:
I think you need to trim after emitting. For example, both of following two lines are comments and should be ignored.
```
# comment
   # comment
```